cmd/template.go

```diff
@@ -0,0 +1,56 @@
+/*----------------------------------------------------------------
+ *  Copyright (c) ThoughtWorks, Inc.
+ *  Licensed under the Apache License, Version 2.0
+ *  See LICENSE in the project root for license information.
+ *----------------------------------------------------------------*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/getgauge/gauge/logger"
+	"github.com/getgauge/gauge/template"
+	"github.com/spf13/cobra"
+)
+
+var (
+	templateCmd = &cobra.Command{
+		Use:     "template [flags] [args]",
+		Short:   "Change template configurations",
+		Long:    `Change template configurations.`,
+		Example: `  gauge template java getgauge/java-template`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if templateList || machineReadable {
+				text, err := template.List(machineReadable)
+				if err != nil {
+					logger.Fatalf(true, err.Error())
+				}
+				logger.Infof(true, text)
+				return
+			}
+			if len(args) == 0 {
+				exit(fmt.Errorf("template command needs argument(s)"), cmd.UsageString())
+			}
+			if len(args) == 1 {
+				text, err := template.Get(args[0])
+				if err != nil {
+					logger.Fatalf(true, err.Error())
+				}
+				logger.Infof(true, text)
+				return
+			}
+			err := template.Update(args[0], args[1])
+			if err != nil {
+				exit(fmt.Errorf("template URL should be a valid link of zip"), cmd.UsageString())
```

Comment:
I am not sure if it reaches here. The following command gives a misleading message.

```
gauge template custom https://example.org/zip
➜  vowels gauge init custom
Initializing template from https://example.org/zip
Compatible language plugin custom is not installed. Installing plugin...
Failed to install plugin 'custom'.
Reason: Invalid plugin. Could not download custom-install.json file.
```

When a link to a zip file that is not available is added it should ideally be

```
gauge template custom https://example.org/zip
➜  vowels gauge init custom
Initializing template from https://example.org/zip
Failed to download template from `https://example.org/zip` please sure that this is a valid Gauge template URI or there are no problems with the network connection.
```

Also when running something that is NOT in the `template.properties` it currently shows a misleading message for example

```
gauge init jaba
Compatible language plugin abcd is not installed. Installing plugin...
Failed to install plugin 'abcd'.
Reason: Invalid plugin. Could not download abcd-install.json file.
```

This should be

```
gauge init jaba
Cannot find a Gauge template for `abcd`
```

Bonus points if this is implemented as

```
gauge init jaba
Cannot find a Gauge template for `abcd`
The most similar template names are
  
   java

```

Reply:
> Also when running something that is NOT in the template.properties it currently shows a misleading message for example

This happens cause when it fails to initialize a template, it tries to install the language runner, and perform an offline initialization. The language runners name is figured out from the template name (`js` in case of `js_simple` or `js_selenium`)

We need to address this as well (not allow an offline init via runner Ref #1639 ),
Also to install the runner after init the runner name should be fetched from `manifest` not from the template name.